### PR TITLE
ChangePendingModeratorStatus

### PR DESCRIPTION
**Pull request by aabdoo23**
#### PR Classification
New feature to handle pending moderator status modification and retrieval.

#### PR Summary
Introduced functionality for managing pending moderators, including status modification and retrieval.
- `AdminController`: Added methods for pending moderator status modification and retrieval.
- `ModifyPendingModeratorStatusCommand` and related classes: Added to support status modification.
- `GetPendingModeratorsQuery` and related classes: Added to support querying pending moderators.
- `IBaseRepository`: Updated to support `Guid` type and added pagination methods.
- `ServiceCollectionExtensions`: Updated to register new services and repositories.

### CODE REVIEW

File: SchedsForums.Application/Commands/Moderators/ModifyPendingModeratorStatus/ModifyPendingModeratorStatusCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                    Username = pendingModerator.Username,
+                    Email = pendingModerator.Email,
+                    PasswordHash = pendingModerator.PasswordHash,
+                    Status = ModeratorStatus.Approved,
+                    CreatedAt = pendingModerator.CreatedAt,
+                    StatusUpdatedAt = DateTime.UtcNow,
+                    StatusUpdatedBy = admin,
+                    Reason = pendingModerator.Reason
+                };
+                await _moderatorRepository.InsertAsync(moderator);
```
Comment (MostafaHgz) [inline SchedsForums.Application/Commands/Moderators/ModifyPendingModeratorStatus/ModifyPendingModeratorStatusCommandHandler.cs:52]: just update the pending moderator to be moderator. no need to create a new one

File: SchedsForums.Application/Commands/Moderators/ModifyPendingModeratorStatus/ModifyPendingModeratorStatusCommandValidator.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        private readonly IPendingModeratorRepository _pendingModeratorRepository;
+        public ModifyPendingModeratorStatusCommandValidator(IPendingModeratorRepository pendingModeratorRepository)
+        {
+            _pendingModeratorRepository = pendingModeratorRepository ??
+                throw new ArgumentNullException(nameof(pendingModeratorRepository));
+
+            RuleFor(x => x.ModeratorId.ToString())
+                .NotEmpty()
+                .MinimumLength(20);
+            //.Must(IsValidModeratorId);
```
Comment (MostafaHgz) [inline SchedsForums.Application/Commands/Moderators/ModifyPendingModeratorStatus/ModifyPendingModeratorStatusCommandValidator.cs]: check commented lines

File: SchedsForums.Infrastructure/Services/CurrentUserService.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+{
+    public class CurrentUserService(IHttpContextAccessor httpContextAccessor) : ICurrentUserService
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor = httpContextAccessor
+                ?? throw new ArgumentNullException(nameof(httpContextAccessor));
+
+        public Guid GetUserId()
+        {
+            var userId = GetClaimValue(ClaimTypes.NameIdentifier);
+            var guid = Guid.Parse(userId);
```
Comment (MostafaHgz) [inline SchedsForums.Infrastructure/Services/CurrentUserService.cs]: return Guid.Parse(userId)

File: SchedsForums.Infrastructure/Services/CurrentUserService.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                ?? throw new InvalidOperationException("Username claim not found");
+        }
+
+        public string GetUserRole()
+        {
+            return GetClaimValue(ClaimTypes.Role)
+                ?? throw new InvalidOperationException("User role claim not found");
+        }
+
+        public bool IsAuthenticated()
```
Comment (MostafaHgz) [inline SchedsForums.Infrastructure/Services/CurrentUserService.cs]: why would you need that? having all these claims means it's authenticated so no need for it. authentication is done by jwt not here

**Review by MostafaHgz — CHANGES_REQUESTED** [review]

File: SchedsForums.Application/Commands/Moderators/ModifyPendingModeratorStatus/ModifyPendingModeratorStatusCommandHandler.cs (hunk 1 above)
Comment (aabdoo23) [reply SchedsForums.Application/Commands/Moderators/ModifyPendingModeratorStatus/ModifyPendingModeratorStatusCommandHandler.cs:52]: "details": "The property 'PendingModerator.UserType' is defined as read-only after it has been saved, but its value has been modified or marked as modified."

            modelBuilder.Entity<Book>().Property<string>("Discriminator").Metadata.IsReadOnlyAfterSave = false;
i found the solution but i don't think i want to od that, i need to keep the classes as separated as possible, i feel like making the UserType mutable would kind of violate that, give me your opinion as well

File: SchedsForums.Application/Commands/Moderators/ModifyPendingModeratorStatus/ModifyPendingModeratorStatusCommandValidator.cs (hunk 2 above)
Comment (aabdoo23) [reply SchedsForums.Application/Commands/Moderators/ModifyPendingModeratorStatus/ModifyPendingModeratorStatusCommandValidator.cs]: after some searching i found out that you can't register async validations in the normal pipeline, so i'll validate it in the command handler

**Pull request merged**